code/datums/components/itempicky.dm

```diff
@@ -5,13 +5,21 @@
 	var/whitelist
 	/// Message shown if you try to pick up an item not in the whitelist
 	var/message = "You don't like %TARGET, why would you hold it?"
+	/// An optional callback we check for overriding our whitelist
+	var/datum/callback/tertiary_condition = null
 
-/datum/component/itempicky/Initialize(whitelist, message)
+/datum/component/itempicky/Initialize(whitelist, message, tertiary_condition)
 	if(!ismob(parent))
 		return COMPONENT_INCOMPATIBLE
 	src.whitelist = whitelist
 	if(message)
 		src.message = message
+	if(tertiary_condition)
+		src.tertiary_condition = tertiary_condition
+
+/datum/component/itempicky/Destroy(force)
+	. = ..()
+	tertiary_condition = null
```

Comment:
I WILL be anal (destroy should be top down because the parent call (could in theory) rely on our type's proc overrides)
```suggestion
	tertiary_condition = null
	return ..()
```